v3/src/utilities/js-utils.ts

```diff
@@ -173,3 +173,15 @@ export function hashStringSets(stringSets: Array<string[]>) {
     // eslint-disable-next-line no-bitwise
     .reduce((acc, hash) => acc ^ hash, 0) // XOR all individual hashes
 }
+
+/*
+ * hashOrderedStringSet()
+ *
+ * returns an order-dependent hash value for a set of strings (e.g. ids).
+ * developed with the help of ChatGPT.
+ */
+export function hashOrderedStringSet(strings: string[]) {
```

Comment:
If we find this isn't working well (speed or accuracy) we could try doing `hashString(JSON.stringify(strings))`

Reply:
I would expect the `JSON.stringify` version to be slower, but it'd be interesting to check.